src/zelos/feeds.py

```diff
@@ -0,0 +1,226 @@
+# Copyright (C) 2020 Zeropoint Dynamics
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+# ======================================================================
+
+"""Implementation of feeds.
+
+Feeds are a way to subscribe to information that is retrieved from a
+dynamic execution of a binary, while respecting the performance
+optimizations that have been requested by the user. This way, a user can
+specify what kinds of information should be collected (each with their
+own performance cost) in a global way, without having to configure
+multiple plugins separately.
+
+To start with, there are different levels of feeds, each increasing in
+amount of verbosity as well as performance cost::
+
+                     -> Verbosity ->
+            None -> Syscalls -> Apis -> Instructions
```

Comment:
Should stdout and stderr be feeds?

Reply:
Based on offline conversation, feeds will likely only be information related to the execution of code, rather than the output of that code.